src/legacy/core_plugins/kibana/public/management/sections/settings/advanced_settings.tsx

```diff
@@ -37,18 +38,35 @@ import {
 } from './components/default_component_registry';
 import { getSettingsComponent } from './components/component_registry';
 
-export class AdvancedSettings extends Component {
-  static propTypes = {
-    config: PropTypes.object.isRequired,
-    query: PropTypes.string,
-    enableSaving: PropTypes.bool.isRequired,
-  };
+interface AdvancedSettingsProps {
+  config: IUiSettingsClient;
+  query: string;
+  enableSaving: boolean;
+}
+
+interface AdvancedSettingsState {
+  footerQueryMatched: boolean;
+  query: IQuery;
+  filteredSettings: Record<string, FieldSetting[]>;
+}
+
+type GroupedSettings = Record<string, FieldSetting[]>;
+
+export class AdvancedSettings extends Component<AdvancedSettingsProps, AdvancedSettingsState> {
+  private settings: FieldSetting[];
+  private groupedSettings: GroupedSettings;
+  private categoryCounts: Record<string, number>;
+  private categories: string[] = [];
 
-  constructor(props) {
+  constructor(props: AdvancedSettingsProps) {
     super(props);
     const { config, query } = this.props;
     const parsedQuery = Query.parse(query ? `ariaName:"${getAriaName(query)}"` : '');
-    this.init(config);
+    this.settings = this.initSettings(config);
```

Comment:
This block is the same as calling `init` isn't it?

Reply:
yes, typescript only recognizes properties set directly in constructor, otherwise I could just call `init`